README.md

```diff
@@ -54,3 +54,51 @@ De API variabelen gebruiken dubbele underscores (`__`) voor .NET configuratie bi
 helm install datamigratie ./charts/datamigratie -f jouw-values.yaml
 ```
 
+## Data Mapping and Transformation Rules
+
+During migration from DET to OpenZaak, certain field values must be transformed to comply with OpenZaak's constraints. The following rules are applied automatically by the `MigrateZaakService`:
+
+### Zaak Field Mappings
+
+| DET Field | OpenZaak Field | Transformation | Max Length |
+|-----------|----------------|----------------|------------|
+| `Omschrijving` | `Omschrijving` | Truncated with "..." suffix if too long | 80 characters |
+| `FunctioneleIdentificatie` | `Identificatie` | No transformation | - |
+| `Startdatum` | `Startdatum` | Formatted as `yyyy-MM-dd` | - |
+| `CreatieDatumTijd` | `Registratiedatum` | Formatted as `yyyy-MM-dd` | - |
+
+### Document Field Mappings
+
+| DET Field | OpenZaak Field | Transformation | Max Length |
+|-----------|----------------|----------------|------------|
+| `Titel` | `Titel` | Truncated with "..." suffix if too long | 200 characters |
+| `Beschrijving` | `Beschrijving` | Truncated with "..." suffix if too long | 1000 characters |
+| `Kenmerk` | `Identificatie` | **Migration fails** if exceeds max length | 40 characters |
+
+### Truncation Behavior
+
+When a field value exceeds its maximum allowed length, the following truncation logic is applied:
+
+1. If the value is within the limit, it is used as-is
+2. If the value exceeds the limit, it is truncated and "..." is appended
+3. The final length (including "...") equals the maximum allowed length
+4. Trailing whitespace is removed before appending "..."
+
+**Example:**
+- Input: `"Hello world"`, Max length: `5`
+- Output: `"He..."` (length = 5)
+
+### Migration Failures
+
+The migration will **fail** for a document if:
+- The `Kenmerk` field exceeds 40 characters (this field cannot be truncated as it serves as a unique identifier)
+
+When this occurs, an error message is returned indicating which document caused the failure and why.
+
+### Implementation Details
+
+All transformation logic is implemented in [MigrateZaakService.cs](Datamigratie.Server/Features/MigrateZaak/MigrateZaakService.cs):
```

Comment:
Voor wie is dit? Lijkt me onmogelijk om dit up to date te houden. Ik zou het stukje Implementation Details weglaten